tests/hash_test.go

```diff
@@ -245,6 +245,67 @@ var _ = Describe("hash", func() {
 		})
 	})
 
+	Context("Will include field names in the hash", func() {
```

Comment:
Super nit: Do you mind changing this context to:

```suggestion
	Context("Hashing object values of same type", func() {
```

We aren't concerned with the implementation detail that field names are included in hash, just that the behavior of unique hashes is enforced.